cli/cli/kurtosis_config/overrides_migrators/config_overrides_migrators.go

```diff
@@ -26,13 +27,87 @@ type configOverridesMigrator = func(uncastedOldConfig interface{}) (interface{},
 // to the bottom each time
 // >>>>>>>>>>>>>>>>>>>>>>>>>>>>> INSTRUCTIONS <<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<
 var AllConfigOverridesMigrators = map[config_version.ConfigVersion]configOverridesMigrator{
+	config_version.ConfigVersion_v4: migrateFromV4,
 	config_version.ConfigVersion_v3: migrateFromV3,
 	config_version.ConfigVersion_v2: migrateFromV2,
 	config_version.ConfigVersion_v1: migrateFromV1,
 	config_version.ConfigVersion_v0: migrateFromV0,
 }
 
 // vvvvvvvvvvvvvvvvvvvvvvv REVERSE chronological order so you don't have to scroll forever vvvvvvvvvvvvvvvvvvvv
+func migrateFromV4(uncastedConfig interface{}) (interface{}, error) {
+	// cast "uncastedConfig" to current version we're upgrading from
+	castedOldConfig, ok := uncastedConfig.(*v4.KurtosisConfigV4)
+	if !ok {
+		return nil, stacktrace.NewError(
+			"Failed to cast old configuration '%+v' to expected configuration struct",
+			uncastedConfig,
+		)
+	}
+
+	var newClusters map[string]*v5.KurtosisClusterConfigV5
+	if castedOldConfig.KurtosisClusters != nil {
+		newClusters = map[string]*v5.KurtosisClusterConfigV5{}
+		for oldClusterName, oldClusterConfig := range castedOldConfig.KurtosisClusters {
+			oldKubernetesConfig := oldClusterConfig.Config
+			oldLogsAggregatorConfig := oldClusterConfig.LogsAggregator
+
+			var newKubernetesConfig *v5.KubernetesClusterConfigV5
+			if oldKubernetesConfig != nil {
+				newKubernetesConfig = &v5.KubernetesClusterConfigV5{
+					KubernetesClusterName:  oldKubernetesConfig.KubernetesClusterName,
+					StorageClass:           oldKubernetesConfig.StorageClass,
+					EnclaveSizeInMegabytes: oldKubernetesConfig.EnclaveSizeInMegabytes,
+					EngineNodeName:         oldKubernetesConfig.EngineNodeName,
+				}
+			}
+
+			var newLogsAggregatorConfig *v5.LogsAggregatorConfigV5
+			if oldLogsAggregatorConfig != nil {
+				newLogsAggregatorConfig = &v5.LogsAggregatorConfigV5{
+					Sinks: oldLogsAggregatorConfig.Sinks,
+				}
+			}
+
+			newGraflokiConfig := &v5.GraflokiConfig{
+				ShouldStartBeforeEngine: false,
```

Comment:
Seems weird that this is configurable. If Grafana can grab the engine logs, wouldn't we always want to start it before the engine so it can capture engine logs too?

Reply:
Grafana is only capturing logs from services in enclaves right now - altho it could one day capture logs.

Grafana / Loki is not a default, its an available option if users want it. The default is the simpler persistent volume /kurtosis service logs setup. 

We give the users the ability to use Grafana / Loki - `grafloki start` `grafloki stop` AND let them make it sticky by having kurtosis always do `grafloki start` before `engine start` when configured.